backend/app/api/ws/chat_namespace.py

```
@@ -322,6 +330,90 @@ async def on_task_join(self, sid: str, data: dict) -> dict:
+                # Full sync: fetch all messages using existing service
+                from app.services.adapters.task_kinds import task_kinds_service
+
+                task_detail = task_kinds_service.get_task_detail(
+                    db, task_id=payload.task_id, user_id=user_id
+                )
+                subtasks_dict = task_detail.get("subtasks")
+                logger.info(
+                    f"[WS] task:join full sync: fetched {len(subtasks_dict) if subtasks_dict else 0} "
+                    f"subtasks for task_id={payload.task_id}"
+                )
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)